lib/web/apps.go

```diff
@@ -442,3 +452,16 @@ func (h *Handler) proxyDNSNames() (dnsNames []string) {
 	}
 	return dnsNames
 }
+
+// proxyPublicAddrs returns a list of public proxy addresses as strings.
+// It collects the addresses from h.cfg.ProxyPublicAddrs. If no addresses
+// are configured, it defaults to the cluster name from h.auth.
+func (h *Handler) proxyPublicAddrs() (proxyPublicAddrs []string) {
+	for _, addr := range h.cfg.ProxyPublicAddrs {
+		proxyPublicAddrs = append(proxyPublicAddrs, addr.String())
```

Comment:
Not sure where do we initially parse the config file but would it be good idea to do `addr.String()` once in the source rather than doing that everywhere?